analytic_engine/src/sst/parquet/async_reader.rs

```diff
@@ -371,39 +434,36 @@ impl<'a> SstReader for Reader<'a> {
     async fn read(
         &mut self,
     ) -> Result<Box<dyn Stream<Item = Result<RecordBatchWithKey>> + Send + Unpin>> {
-        self.init_if_necessary().await?;
+        let mut streams = self.read_parallelly(1).await?;
+        assert_eq!(streams.len(), 1);
+        let stream = streams.pop().expect("impossible to fetch no stream");
 
-        let stream = self.fetch_record_batch_stream().await?;
-        let row_projector = self.row_projector.take().unwrap();
-        let row_projector = ArrowRecordBatchProjector::from(row_projector);
-
-        let storage_format_opts = self
-            .meta_data
-            .as_ref()
-            // metadata must be inited after `init_if_necessary`.
-            .unwrap()
-            .custom()
-            .storage_format_opts
-            .clone();
-
-        Ok(Box::new(RecordBatchProjector::new(
-            self.path.to_string(),
-            stream,
-            row_projector,
-            storage_format_opts,
-        )))
+        Ok(stream)
     }
 }
 
 struct RecordBatchReceiver {
-    rx: Receiver<Result<RecordBatchWithKey>>,
+    rx_group: Vec<Receiver<Result<RecordBatchWithKey>>>,
+    cur_rx_idx: usize,
 }
 
 impl Stream for RecordBatchReceiver {
     type Item = Result<RecordBatchWithKey>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        self.as_mut().rx.poll_recv(cx)
+        let cur_rx_idx = self.cur_rx_idx;
+        // `cur_rx_idx` is impossible to be out-of-range, because it is got by round
+        // robin.
+        let cur_rx = self.rx_group.get_mut(cur_rx_idx).unwrap();
+        let poll_result = cur_rx.poll_recv(cx);
+
+        match poll_result {
```

Comment:
Add comments why only advance idx in Ready

Reply:
Done.